### PR TITLE
build(deps): bump @testing-library/react-native to 12.4.3

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves [] 

### Description

bump @testing-library/react-native to 12.4.3

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- bump @testing-library/react-native to 12.4.3 - gkartalis

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

File: ios/Podfile.lock [hunk 1]
```diff
@@ -1308,7 +1308,7 @@ SPEC CHECKSUMS:
   appcenter-core: e192ea8b373bebd3e44998882b43311078bd7dda
   AppCenterReactNativeShared: 01df23849b1c3c6eb8c4049f54322635650e98f0
   Base64: cecfb41a004124895a7bcee567a89bae5a89d49b
-  boost: 57d2868c099736d80fcd648bf211b4431e51a558
+  boost: 7dcd2de282d72e344012f7d6564d024930a6a440
```
Comment (MounirDhahri) [inline ios/Podfile.lock:1311]: is this intentional?

**Review by MounirDhahri — APPROVED** [review]
LGTM

**Review by MrSltun — APPROVED** [review]

File: ios/Podfile.lock (hunk 1 above)
Comment (gkartalis) [reply ios/Podfile.lock:1311]: yep, I should have mentioned that ~ tried to do `➜ cd ios && bundle exec pod update boost --no-repo-update && cd ..` but didn't fix it so I committed it 🤔

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (bump @testing-library/react-native to 12.4.3 - gkartalis)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 58fa52f9a7a1840ff880f9675c500bdb52dd0bf3
</p>

File: ios/Podfile.lock (hunk 1 above)
Comment (gkartalis) [reply ios/Podfile.lock:1311]: I think this re-appeared due to the latest rn upgrade bumps that we did

Comment (gkartalis) [reply ios/Podfile.lock:1311]: https://github.com/facebook/react-native/releases/tag/v0.72.9

**Pull request merged**